### PR TITLE
Updating Microsoft.Insights swagger spec to include api-version for Management groups

### PR DESCRIPTION
**Pull request by admukhty**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (MyronFanQiu) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:236]: Isn't category required?

File: specification/monitor/resource-manager/readme.md
Comment (MyronFanQiu) [inline specification/monitor/resource-manager/readme.md:34]: Please modify readme.python.md as well

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:236]: Yes, thanks will update it

File: specification/monitor/resource-manager/readme.md
Comment (admukhty) [reply specification/monitor/resource-manager/readme.md:34]: Updated

File: specification/monitor/resource-manager/readme.md
Comment (admukhty) [inline specification/monitor/resource-manager/readme.md:41]: @MyronFanQiu Please review if this is the right way to add a new api-version.

Would the better way be to add a new package with all the existing api-versions from the previous package, and only add the new api-version to that. so something like this:

``` yaml
title: MonitorClient
description: Monitor Management Client
openapi-type: arm
tag: package-2020-06
```

 

### Tag: package-2020-06

 

These settings apply only when `--tag=package-2020-06` is specified on the command line.

 

```yaml $(tag) == 'package-2020-06'
input-file:
- Microsoft.Insights/stable/2015-04-01/autoscale_API.json
- Microsoft.Insights/stable/2015-04-01/operations_API.json
- Microsoft.Insights/stable/2016-03-01/alertRulesIncidents_API.json
- Microsoft.Insights/stable/2016-03-01/alertRules_API.json
- Microsoft.Insights/stable/2016-03-01/logProfiles_API.json
- Microsoft.Insights/preview/2017-05-01-preview/diagnosticsSettings_API.json
- Microsoft.Insights/preview/2017-05-01-preview/diagnosticsSettingsCategories_API.json
- Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
- Microsoft.Insights/stable/2017-04-01/activityLogAlerts_API.json
- Microsoft.Insights/stable/2015-04-01/activityLogs_API.json
- Microsoft.Insights/stable/2015-04-01/eventCategories_API.json
- Microsoft.Insights/stable/2015-04-01/tenantActivityLogs_API.json
- Microsoft.Insights/stable/2018-01-01/metricDefinitions_API.json
- Microsoft.Insights/stable/2018-01-01/metrics_API.json
- Microsoft.Insights/preview/2017-11-01-preview/baseline_API.json
- Microsoft.Insights/preview/2017-11-01-preview/calculateBaseline_API.json
- Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json
- Microsoft.Insights/stable/2018-03-01/metricAlert_API.json
- Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
- Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
- Microsoft.Insights/preview/2018-11-27-preview/vmInsightsOnboarding_API.json
- Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json
- Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
```


If this is the right way, what should be the change in the readme.python.md file?

**Comment by NullMDR** [comment]
@changlong-liu Could you please help to confirm why do we have so much breaking change in python SDK?

**Comment by admukhty** [comment]
@changlong-liu Please help to review this PR.

Please review the comments on the PR for the readme.md file. What is the best way to add a new api-version to the swagger spec?

**Comment by changlong-liu** [comment]
Hi @NullMDR , @admukhty ,
The Python breaking change is not related to this PR. I think it's fine for Python to merge this PR.

Hi @NullMDR ,
This is related to the Track2 release last week, and I just created this PR for Python: https://github.com/Azure/azure-sdk-for-python/pull/12211
The Python code report will turn to fine after that PR is merge.

File: specification/monitor/resource-manager/readme.md
Comment (admukhty) [reply specification/monitor/resource-manager/readme.md:41]: @changlong-liu  can you review this comment?

Comment (changlong-liu) [reply specification/monitor/resource-manager/readme.md:41]: Hi @admukhty , I think it's fine so far for Python in readme.md.  And the change in readme.python.md should be just in the same way what you do in this PR.

**Review by mmyyrroonn — APPROVED** [review]

**Comment by admukhty** [comment]
@MyronFanQiu<mailto:notifications@github.com> please help merge this pull request.
It seems like I am not authorized to merge it.
[cid:image002.jpg@01D64CCE.DEAE7F00]

Thanks,
Aditi

**Comment by MyronFanQiu** [comment]
> @MyronFanQiu<mailto:notifications@github.com> please help merge this pull request. It seems like I am not authorized to merge it. [cid:image002.jpg@01D64CCE.DEAE7F00] Thanks, Aditi From: MyronFanQiu <notifications@github.com> Sent: Saturday, June 27, 2020 8:33 PM To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com> Cc: Aditi Mukhtyar <Aditi.Mukhtyar@microsoft.com>; Mention <mention@noreply.github.com> Subject: Re: [Azure/azure-rest-api-specs] Updating Microsoft.Insights swagger spec to include api-version for Management groups (#9938) @MyronFanQiu approved this pull request. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F9938%23pullrequestreview-438740384&data=02%7C01%7CAditi.Mukhtyar%40microsoft.com%7C3db77d42427745b8ad0808d81b13eb7f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637289119665590246&sdata=jd%2FNnYFvCzaDp%2B5zO4Bs%2FBu6SahiHaAiGLvbM7ERbOQ%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAQBXUIK5MCAPESLEOCKVCFTRY22VXANCNFSM4OF5VH5Q&data=02%7C01%7CAditi.Mukhtyar%40microsoft.com%7C3db77d42427745b8ad0808d81b13eb7f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637289119665600240&sdata=vn7Vt8Dg6%2FvAlgdloCShupZuC7bmirYXhOal0KuZnNs%3D&reserved=0>.

we still need ARM team's review and approval.

**Comment by admukhty** [comment]
@MyronFanQiu<mailto:notifications@github.com> can you help add someone from the ARM team to review this?

I see it is has been waiting for over a week.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (pilor) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:216]: this resource should not have a location since it is proxyOnly

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (pilor) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:241]: Is the diagnosticSettings definition different at MG level than other levels?  Why can't it use the same definition used for normal diagnostic settings?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (pilor) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:161]: this should include "x-ms-pageable": true even if you don't support nextLinks today.  Prevents breaking changes if you add them in the future

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json
Comment (pilor) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json:30]: type should not be null in examples

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json:30]: @pilor what should it be then?
This is the output I get when I test the API in Postman

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:161]: @pilor  where exactly should this be added?

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:216]: Removed

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:241]: The definition is essentially the same. 
The path is different -- so a normal diagnostic setting has <resourceUri>/providers/microsoft.insights/diagnosticSettings
Management group will have "/providers/microsoft.management/managementGroups/<mgId>/providers/microsoft.insights/diagnosticSettings"

My understanding is that if I am introducing a new api-version, and a new file I would need to define this again (Please see how subscriptionDiagnosticSettings has been done-- https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json)

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json
Comment (pilor) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json:30]: If your API is outputting "Type" as null it needs to be fixed.  This is your resource type "Microsoft.Insights/diagnosticSettings".

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (pilor) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:161]: search the repo for the below snippet.  It goes anywhere in the GET collection operation

```
"x-ms-pageable":` {
          "nextLinkName": null
        }
```

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json
Comment (pilor) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/managementGroupDiagnosticSettings_API.json:216]: this property is still here

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json
Comment (admukhty) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2020-01-01-preview/examples/createOrUpdateManagementGroupDiagnosticSetting.json]: Made the change here. Will also update the API to emit it correctly

**Review by pilor — APPROVED** [review]

**Comment by admukhty** [comment]
@MyronFanQiu<mailto:notifications@github.com> the ARM team has reviewed and approved the PR.
Please help merge it.

Thanks,
Aditi

**Pull request merged**
